src/test/fuzz/txrequest.cpp

```diff
@@ -0,0 +1,363 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <primitives/transaction.h>
+#include <random.h>
+#include <txrequest.h>
+#include <test/fuzz/fuzz.h>
+#include <crypto/common.h>
+#include <crypto/siphash.h>
+
+#include <bitset>
+#include <cstdint>
+#include <queue>
+#include <vector>
+
+namespace {
+
+constexpr int MAX_TXHASHES = 16;
+constexpr int MAX_PEERS = 16;
+
+//! Randomly generated GenTxids used in this test (length is MAX_TX).
+uint256 TXHASHES[MAX_TXHASHES];
+
+/** Precomputed random durations (positive and negative, each ~exponentially distributed). */
+std::chrono::microseconds DELAYS[256];
+
+struct Initializer
+{
+    Initializer()
+    {
+        // Use deterministic RNG to fill in txids and delays.
+        // Non-determinism hurts fuzzing.
+        FastRandomContext rng(true);
+        for (int txhash = 0; txhash < MAX_TXHASHES; txhash += 1) {
+            do {
+                TXHASHES[txhash] = rng.rand256();
+            } while (*(TXHASHES[txhash].begin() + 31) != txhash || *(TXHASHES[txhash].begin()) != txhash);
```

Comment:
8d0dd46b4fcdf4c664747e638eda046f8d51079c

What is the reason for these conditions? Why does any `rand256()` not suit?

Reply:
It does. This just helped debugging as it meant I could look at a hex txid and know which #txhash it was. Removed it.